mesh/silvair-io.c

```diff
@@ -132,49 +135,105 @@ struct silvair_keep_alive_cmd_pld {
 } __packed;
 
 
+static bool buffer_write(struct l_io *l_io, void *user_data)
+{
+    struct silvair_io *io = user_data;
+
+    int consumed = l_ringbuf_write(io->l_out_ringbuf, l_io_get_fd(l_io));
+
+    if (consumed >= 0) {
+        l_info("Data from buffer have been written.");
+    }
+
+    return (bool)l_ringbuf_len(io->l_out_ringbuf);
+}
+
 static bool simple_write(struct silvair_io *io,
 				uint8_t *buf,
 				size_t size)
 {
-	int w = write(l_io_get_fd(io->l_io), buf, size);
+    int consumed = 0;
+    ssize_t buffered;
+
+    size_t space_left = l_ringbuf_capacity(io->l_out_ringbuf) - l_ringbuf_len(io->l_out_ringbuf);
+
+    /* To be able send whole SLIP frame, there have to be space for whole frame in the buffer. */
+    if (space_left < size) {
+        l_info("Write failed. Not enough space in the buffer.");
+        return false;
+    }
 
-	return (w > 0 && (size_t)w == size);
+    consumed = write(l_io_get_fd(io->l_io), buf, size);
```

Comment:
Why do you write to fd directly here? Why not pump everything through the ringbuf?